agent/api/task/task.go

```diff
@@ -984,70 +987,77 @@ func (task *Task) AddFirelensContainerBindMounts(firelensConfigType string, host
 	return nil
 }
 
-// BuildCNIConfig constructs the cni configuration from eni
-func (task *Task) BuildCNIConfig() (*ecscni.Config, error) {
-	if !task.isNetworkModeVPC() {
-		return nil, errors.New("task config: task has no ENIs associated with it, unable to generate cni config")
+// BuildCNIConfig builds a list of CNI network configurations for the task.
+// If includeIPAMConfig is set to true, the list also includes the bridge IPAM configuration.
+func (task *Task) BuildCNIConfig(includeIPAMConfig bool, cniConfig *ecscni.Config) (*ecscni.Config, error) {
```

Comment:
I did not see any unit tests against this function, can we add some test cases for it?

Reply:
The tests for this exist in docker_task_engine_test.go (see https://github.com/aws/amazon-ecs-agent/pull/2173/files#diff-3aedbbecbf5a79074e46dafde46c1536R1235). I've moved them now to `task_test.go`